cms/io/rpc.py

```diff
@@ -478,8 +479,12 @@ def _connect(self):
 
         """
         try:
-            sock = gevent.socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-            sock.connect(self.remote_address)
+            # Find the family (AF_INET or AF_INET6) of the IPv4/6 address.
+            addresses = gevent.socket.getaddrinfo(
+                self.remote_address.ip, self.remote_address.port)
+            family, *rest, sockaddr = addresses[0]
```

Comment:
It is slightly worrying to take only the first returned address. It would be more prudent to try all of them. From Linux `getaddrinfo(3)`:
> Normally, the application should try using the addresses in the order in which they are returned.

Can you please implement a loop that tries all addresses in succession? A related example is available at https://docs.python.org/3/library/socket.html#example.

Reply:
Done, tell me if you like the loop like that.